subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesGroovyDSLDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,1930 @@
+
+        [bundles]
+        groovy = ["groovy-core", "groovy-json", "groovy-nio"]
+        """.stripIndent(8)
+
+        expect:
+        succeeds 'checkConfiguration'
+
+        where:
+        suiteDesc           | suiteName   | suiteDeclaration
+        'the default suite' | 'test'      | 'test'
+        'a custom suite'    | 'integTest' | 'integTest(JvmTestSuite)'
+    }
+
+    def 'can add dependencies using a Version Catalog with a hierarchy of aliases to #suiteDesc'() {
+        given:
+        buildFile << """
+        plugins {
+          id 'java-library'
+        }
+
+        ${mavenCentralRepository()}
+
+        testing {
+            suites {
+                $suiteDeclaration {
+                    dependencies {
+                        implementation libs.commons
+                        implementation libs.commons.collections
+                        runtimeOnly libs.commons.io
+                        runtimeOnly libs.commons.io.csv
+                    }
+                }
+            }
+        }
+
+        tasks.named('check') {
+            dependsOn testing.suites.$suiteName
+        }
+
+        tasks.register('checkConfiguration') {
+            dependsOn $suiteName
+            doLast {
+                def ${suiteName}CompileClasspathFileNames = configurations.${suiteName}CompileClasspath.files*.name
+                def ${suiteName}RuntimeClasspathFileNames = configurations.${suiteName}RuntimeClasspath.files*.name
+
+                assert ${suiteName}CompileClasspathFileNames.containsAll('commons-lang3-3.12.0.jar', 'commons-collections4-4.4.jar')
+                assert ${suiteName}RuntimeClasspathFileNames.containsAll('commons-lang3-3.12.0.jar', 'commons-collections4-4.4.jar', 'commons-io-2.11.0.jar', 'commons-csv-1.9.0.jar')
+            }
+        }
+        """
+
+        versionCatalog = file('gradle', 'libs.versions.toml') << """
+        [versions]
+        commons-lang = "3.12.0"
+        commons-collections = "4.4"
+        commons-io = "2.11.0"
+        commons-io-csv = "1.9.0"
+
+        [libraries]
+        commons = { group = "org.apache.commons", name = "commons-lang3", version.ref = "commons-lang" }
+        commons-collections = { group = "org.apache.commons", name = "commons-collections4", version.ref = "commons-collections" }
+        commons-io = { group = "commons-io", name = "commons-io", version.ref = "commons-io" }
+        commons-io-csv = { group = "org.apache.commons", name = "commons-csv", version.ref = "commons-io-csv" }
+        """.stripIndent(8)
+
+        expect:
+        succeeds 'checkConfiguration'
+
+        where:
+        suiteDesc           | suiteName   | suiteDeclaration
+        'the default suite' | 'test'      | 'test'
+        'a custom suite'    | 'integTest' | 'integTest(JvmTestSuite)'
+    }
+
+    def 'can add dependencies using a Version Catalog defined programmatically to #suiteDesc'() {
```

Comment:
I don't think these are any different from the ones defined in a TOML, so we shouldn't need to test them separately, right @octylFractal ?

Reply:
This was a per-existing test.  I think there's a little value in explicitly demonstration and testing either way here, though I agree each should end up functioning the same internally.  We're cataloging the possible syntax here - that's as important a goal as making sure functionality continues to work.